src/Resources/ResourceManager/SdkModels/Deployments/PSDeploymentWhatIfCmdletParameters.cs

```diff
@@ -0,0 +1,102 @@
+﻿namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels.Deployments
+{
+    using System;
+    using System.Collections;
+    using System.IO;
+    using Commands.Common.Authentication.Abstractions;
+    using Management.ResourceManager.Models;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using Properties;
+    using WindowsAzure.Commands.Common;
+
+    public class PSDeploymentWhatIfCmdletParameters
```

Comment:
>PSDeploymentWhatIfCmdletParameters [](start = 17, length = 34)

do we need this new class? can we reuse the existing one?

Reply:
Yes, we'll need a new parameters class anyway. The rationales:
- The ScopeType is `DeploymentWhatIfScopeType` (Sub & RG) not `ScopeType`.
- It takes a WhatIfResultFormat parameter.

However I think it's a good idea to inherit from the existing parameter class and use the `new modifier` to change the ScopeType property, so that we can reuse other properties.